### PR TITLE
Cavatica Widget in File Repo Sidebar

### PR DESCRIPTION
**Pull request by joneubank**
Providing Cavatica interaction in File Repo sidebar.

 * Connect to Cavatica button if not connected
 * Project list if Connected
 * Create a project
 * Copy files to project

### CODE REVIEW

File: src/common/injectGlobals.js [hunk 1]
```diff
@@ -19,9 +19,12 @@ if (personaApiOverride) {
   global.log('warning: using persona api override');
 }
 
+export const arrangerProjectId = process.env.REACT_APP_PROJECT_ID;
```
Comment (Jephuff) [inline src/common/injectGlobals.js]: wanna replace `process.env.REACT_APP_PROJECT_ID` with this in FileRepo.js?

File: src/components/FileRepoSidebar.js [hunk 2]
```diff
@@ -178,27 +178,6 @@ export default ({ projectId, style, sqon, ...props }) => (
...
-      <img
-        alt=""
-        src={cavaticaLogo}
-        css={`
-          width: 28px;
-          margin-right: 7px;
-        `}
-      />Export files to Cavatica
-    </button>
+    <CavaticaSidebar {...props} />
```
Comment (Jephuff) [inline src/components/FileRepoSidebar.js]: Could you rename this to something like `CavaticaExportButton`. the external link could be moved to be a sibling here.

File: src/services/cavatica.js [hunk 3]
```diff
@@ -80,16 +82,70 @@ export const getProjects = async () => {
...
 };
+
+/**
+ * Requires Cavatica File IDs as returned from convertGen3FileIds
+ *
+ * project is the project id, not name (ie. userId/projectName)
+ * ids is an array of strings of the ids to copy
+ */
+export const copyFiles = async ({ project, ids }) => {
+  const { data: { items } } = await ajax.post(cavaticaApiRoot, {
```
Comment (alex-wilmer) [inline src/services/cavatica.js]: if `ajax.post` is unsuccesful, will this crash trying to access `data.items` ?

File: src/components/cavatica/CavaticaAddProject.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            `}
+          >
+            Create a new Cavatica Project
+          </div>
+          <div css={``}>Enter a name for your project:</div>
+          <input
+            className="textInput"
+            type="text"
+            placeholder="Project Name"
+            onChange={e => setProjectName(e.target.value)}
```
Comment (Jephuff) [inline src/components/cavatica/CavaticaAddProject.js]: Might be a good idea to add `value={projectName}` incase we want to populate that down the line.

File: src/components/cavatica/CavaticaAddProject.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          />
+        </div>
+        <div className="footer">
```
Comment (Jephuff) [inline src/components/cavatica/CavaticaAddProject.js]: Not sure if you saw it, but in Modal/index.js there is a `ModalFooter` component. I think you should be able to use that here

File: src/components/cavatica/CavaticaSidebar.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    :disabled {
+      cursor: default;
+      color: ${theme.greyScale2}
+      background-color: ${theme.greyScale5}
+    }
+  }
+`;
+
+const CavaticaSidebar = ({ state, theme, addingProject, setAddingProject, ...props }) => {
+  let content = <CavaticaConnectButton styles={styles(theme)} />;
```
Comment (Jephuff) [inline src/components/cavatica/CavaticaSidebar.js]: you can pass this to className or css (`css={styles(theme)}`or `className={styles(theme)}`) and just pass className along inside the `CavaticaConnectButton` component.

File: src/icons/DoubleChevronLeftIcon.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+
+export default ({ height, width, fill = '#fff', style, className, alt = '', ...props }) => {
+  return (
+    <img
+      src={`data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 99.2 90.93"><defs><style>.cls-1{fill:${fill};}</style></defs><title>icon-double-chevron-left-grey</title><path class="cls-1" d="M38,3.05,3,38A10.55,10.55,0,0,0,3,52.88l35,35A10.42,10.42,0,0,0,52.78,73.14L25.1,45.46,52.78,17.79A10.42,10.42,0,0,0,38,3.05"/><path class="cls-1" d="M81.42,3.05l-35,35a10.55,10.55,0,0,0,0,14.83l35,35A10.42,10.42,0,0,0,96.15,73.14L68.47,45.46,96.15,17.79A10.42,10.42,0,0,0,81.42,3.05"/></svg>`}
```
Comment (Jephuff) [inline src/icons/DoubleChevronLeftIcon.js]: You can import svgs . 
if you put this in a file call `DoubleChevronLeftIcon.svg` you can do 
```
import doubleChevronLeftIcon from './DoubleChevronLeftIcon.svg'
<img src={doubleChevronLeftIcon}/>
```

File: src/services/cavatica.js (hunk 3 above)
Comment (joneubank) [reply src/services/cavatica.js]: More than likely. Whats the recommended approach for handling failure here? 

ajax.post().then().catch() ?

Comment (alex-wilmer) [reply src/services/cavatica.js]: you can use `try / catch` with async calls. i think that's the standard way a la https://github.com/overture-stack/arranger/blob/master/modules/server/src/index.js#L42

File: src/common/injectGlobals.js (hunk 1 above)
Comment (joneubank) [reply src/common/injectGlobals.js]: I don't follow. This is a global constant from our .env file, why would it be setup differently than the other .env variables?

File: src/components/FileRepoSidebar.js (hunk 2 above)
Comment (joneubank) [reply src/components/FileRepoSidebar.js]: Its not just the button though, this wraps the project list and project adder... its the whole sidebar widget removed from the FileRepoSidebar component, just included inside.

File: src/components/cavatica/CavaticaAddProject.js (hunk 5 above)
Comment (joneubank) [reply src/components/cavatica/CavaticaAddProject.js]: Formatting is significantly different, I could try it didn't look promising.. functionality is very similar though, so long as its not tied into the modal state...

File: src/icons/DoubleChevronLeftIcon.js (hunk 7 above)
Comment (joneubank) [reply src/icons/DoubleChevronLeftIcon.js]: I modify the svg with ${fill} in the fill component. This is done to allow color control. Modeled after existing icons.

File: src/common/injectGlobals.js (hunk 1 above)
Comment (Jephuff) [reply src/common/injectGlobals.js]: we are using ` process.env.REACT_APP_PROJECT_ID` directly in FileRepo.js,  just now that you added it here you could update it to use this `arrangerProjectId` value instead.

File: src/icons/DoubleChevronLeftIcon.js (hunk 7 above)
Comment (alex-wilmer) [reply src/icons/DoubleChevronLeftIcon.js]: then you can mess with any attributes, normal react props style

File: src/components/FileRepoSidebar.js (hunk 2 above)
Comment (Jephuff) [reply src/components/FileRepoSidebar.js]: `CavaticaExportWidget` maybe? 
sidebar is confusing, cause it's not a sidebar :D

File: src/components/cavatica/CavaticaAddProject.js (hunk 5 above)
Comment (Jephuff) [reply src/components/cavatica/CavaticaAddProject.js]: If it's different don't worry about it. Just wanted to mention incase it was the same and you just hadn't seen it. It's hiding :)

File: src/common/injectGlobals.js (hunk 1 above)
Comment (joneubank) [reply src/common/injectGlobals.js]: OH, understood now. Will update.

File: src/components/FileRepoSidebar.js (hunk 2 above)
Comment (joneubank) [reply src/components/FileRepoSidebar.js]: Yes. Will rename as Widget. I like Widget.

File: src/icons/DoubleChevronLeftIcon.js (hunk 7 above)
Comment (joneubank) [reply src/icons/DoubleChevronLeftIcon.js]: thats way cooler. We have a handful of icons to remake this way then.

File: src/components/FileRepoSidebar.js (hunk 2 above)
Comment (joneubank) [reply src/components/FileRepoSidebar.js]: Renamed as suggested.

File: src/components/cavatica/CavaticaSidebar.js (hunk 6 above)
Comment (joneubank) [reply src/components/cavatica/CavaticaSidebar.js]: Tried this but couldn't get the styles to apply correctly. Would like to review this setup with you next week.

Comment (joneubank) [reply src/components/cavatica/CavaticaSidebar.js]: Just keeping it this way for now since it is working and about the same code wise.

**Comment by joneubank** [comment]
Leaving the icons as written since its consistent with existing svg implementations. We can edit the whole set as a separate ticket.

Addressed all other comments.

**Comment by joneubank** [comment]
Ready for another review and merge. Thanks!

**Pull request merged**